hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirStatAndListingOp.java

```diff
@@ -262,13 +262,24 @@ private static DirectoryListing getListing(FSDirectory fsd, INodesInPath iip,
             needLocation, false);
         listingCnt++;
         if (listing[i] instanceof HdfsLocatedFileStatus) {
-            // Once we  hit lsLimit locations, stop.
+            // Once we hit lsLimit locations, stop.
             // This helps to prevent excessively large response payloads.
-            // Approximate #locations with locatedBlockCount() * repl_factor
             LocatedBlocks blks =
                 ((HdfsLocatedFileStatus)listing[i]).getLocatedBlocks();
-            locationBudget -= (blks == null) ? 0 :
-               blks.locatedBlockCount() * listing[i].getReplication();
+            if (blks != null) {
+              ErasureCodingPolicy ecPolicy =
+                  listing[i].getErasureCodingPolicy();
+              if (ecPolicy != null && !ecPolicy.isReplicationPolicy()) {
+                // Approximate #locations with locatedBlockCount() *
+                // internalBlocksNum.
+                locationBudget -= blks.locatedBlockCount() *
+                    (ecPolicy.getNumDataUnits() + ecPolicy.getNumParityUnits());
+              } else {
+                // Approximate #locations with locatedBlockCount() * repl_factor
```

Comment:
The current comment is OK for me, let's wait what @ayushtkn will think about. BTW, `repl_factor`, this name is very c/c++ style-ly,  what about `replicationFactor`? Suggest add period at the end of the sentence.

Reply:
Should be fine, good to change and use replicationFactor in the comment as you suggested